src/components/application_manager/src/commands/mobile/on_button_press_notification.cc

```diff
@@ -105,21 +106,24 @@ void OnButtonPressNotification::Run() {
       LOG4CXX_WARN_EXT(logger_, "Null pointer to subscribed app.");
       continue;
     }
-
-    //Send ButtonPress notification only in HMI_FULL or HMI_LIMITED mode
-    if ((mobile_api::HMILevel::HMI_FULL != subscribed_app->hmi_level()) &&
-        (mobile_api::HMILevel::HMI_LIMITED != subscribed_app->hmi_level())) {
-      LOG4CXX_WARN_EXT(logger_, "OnButtonPress in HMI_BACKGROUND or NONE");
-      continue;
+    
+    if(app.valid()) {
+      //Send ButtonPress notification only in HMI_FULL or HMI_LIMITED mode
+      if ((mobile_api::HMILevel::HMI_FULL != subscribed_app->hmi_level()) &&
+          (mobile_api::HMILevel::HMI_LIMITED != subscribed_app->hmi_level())) {
+        LOG4CXX_WARN_EXT(logger_, "OnButtonPress notification is allowed only"
+                   << "in FULL or LIMITED hmi level when AppID is present");
+        continue;
+      }
+      //if "appID" is present  send it to Named app , only if its FULL or LIMITED
+      if( app == subscribed_app) {
```

Comment:
@madhuy This is just pointer comparision. So it will comprae only adresses. Are you sure that this is what you want to compare?